staging/src/k8s.io/apiserver/pkg/util/peerproxy/gv_exclusion_manager_test.go

```diff
@@ -0,0 +1,539 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peerproxy
+
+import (
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	apidiscoveryv2 "k8s.io/api/apidiscovery/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetExclusionSet(t *testing.T) {
+	tests := []struct {
+		name       string
+		activeGVs  map[schema.GroupVersion]struct{}
+		deletedGVs map[schema.GroupVersion]time.Time
```

Comment:
Is it possible for `deletedGVs` to contain GVs not in `activeGVs`?  Any cases we should test for completeness?

Reply:
yes it is possible. Thats what the "only deleted GVs" case is covering - it has GVs in the `deletedGVs` only and `activeGVs` is nil. 

I added a new case for "same GV in both active and deleted" which can also be a valid state. Ultimately for all these cases, we want `getExclusionSet()` to return union of both `activeGVs` & `deletedGVs`